ggml-cuda.cu

```diff
@@ -1225,7 +1225,7 @@ static void ggml_cuda_op_mul_mat_cublas(
 
     // the main device has a larger memory buffer to hold the results from all GPUs
     // ldc == nrows of the matrix that cuBLAS writes into
-    int ldc = id == ctx.device ? ne0 : row_diff;
+    int64_t ldc = id == ctx.device ? ne0 : row_diff;
```

Comment:
It  would be great to update the PR description to summarize why we upcasted all int params to int64 in this cntext.

Reply:
@phymbert I reverted that one in https://github.com/dranger003/llama.cpp/commit/835d702e56e6ea50d0fe803b56967291194d5e8d. My level of knowledge of these is no where near to be on par with those that created the code and so I definitely rely on your reviews. I looked at some of the values through the debugger but since we have so many overflowing I had to change them in batches, so this means I most likely changed some that don't need to be changed. Hopefully this makes some sense.

Reply:
Thanks for the explanation, please raise with @ggerganov as I am out of the subject regarding CommandR+

Reply:
Thanks, I opened a PR as a follow-up (https://github.com/ggerganov/llama.cpp/pull/6563)